platform/api/oci/client_test.go

```diff
@@ -0,0 +1,182 @@
+	"strings"
+	"testing"
+
+	"github.com/coreos/mantle/auth"
+)
+
+func GetDefaultAuthProfile(t *testing.T) auth.OCIProfile {
+	exampleRSAPrivateKey := `-----BEGIN RSA PRIVATE KEY-----
+MIICXgIBAAKBgQDCFENGw33yGihy92pDjZQhl0C36rPJj+CvfSC8+q28hxA161QF
+NUd13wuCTUcq0Qd2qsBe/2hFyc2DCJJg0h1L78+6Z4UMR7EOcpfdUE9Hf3m/hs+F
+UR45uBJeDK1HSFHD8bHKD6kv8FPGfJTotc+2xjJwoYi+1hqp1fIekaxsyQIDAQAB
+AoGBAJR8ZkCUvx5kzv+utdl7T5MnordT1TvoXXJGXK7ZZ+UuvMNUCdN2QPc4sBiA
+QWvLw1cSKt5DsKZ8UETpYPy8pPYnnDEz2dDYiaew9+xEpubyeW2oH4Zx71wqBtOK
+kqwrXa/pzdpiucRRjk6vE6YY7EBBs/g7uanVpGibOVAEsqH1AkEA7DkjVH28WDUg
+f1nqvfn2Kj6CT7nIcE3jGJsZZ7zlZmBmHFDONMLUrXR/Zm3pR5m0tCmBqa5RK95u
+412jt1dPIwJBANJT3v8pnkth48bQo/fKel6uEYyboRtA5/uHuHkZ6FQF7OUkGogc
+mSJluOdc5t6hI1VsLn0QZEjQZMEOWr+wKSMCQQCC4kXJEsHAve77oP6HtG/IiEn7
+kpyUXRNvFsDE0czpJJBvL/aRFUJxuRK91jhjC68sA7NsKMGg5OXb5I5Jj36xAkEA
+gIT7aFOYBFwGgQAQkWNKLvySgKbAZRTeLBacpHMuQdl1DfdntvAyqpAZ0lY0RKmW
+G6aFKaqQfOXKCyWoUiVknQJAXrlgySFci/2ueKlIE1QqIiLSZ8V8OlpFLRnb1pzI
+7U1yQXnTAEFYM560yJlzUpOb1V4cScGd365tiSMvxLOvTA==
+-----END RSA PRIVATE KEY-----`
+	exampleTenancy := "ocid1.tenancy.oc1..aaaaaaaaba3pv6wkcr4jqae5f15p2b2m2yt2j6rx32uzr4h25vqstifsfdsq"
+	exampleUser := "ocid1.user.oc1..aaaaaaaat5nvwcna5j6aqzjcaty5eqbb6qt2jvpkanghtgdaqedqw3rynjq"
+
+	block, _ := pem.Decode([]byte(exampleRSAPrivateKey))
+	if block == nil {
+		t.Fatalf("failed to parse PEM block")
+	}
+
+	priv, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		t.Fatalf("failed to parse RSA private key: %v", err)
+	}
+
+	profile := auth.OCIProfile{
+		PrivateKey: priv,
+		TenancyID:  exampleTenancy,
+		UserID:     exampleUser,
+	}
+
+	return profile
+}
+
+func TestAuthOCIInjectMissingFields(t *testing.T) {
+	profile := GetDefaultAuthProfile(t)
+
+	err := auth.InjectMissingFields(&profile)
+	if err != nil {
+		t.Fatalf("injecting missing fields: %v", err)
+	}
+
+	// if you're wondering why despite the rest of the default values matching
+	// in this test to the values in signer_test except for the fingerprint
+	// in the KeyID, it's that the Key they give you in the example on the
+	// oracle docs has a different fingerprint than the actual key will produce
+	// this fingerprint was generated via the following command against the
+	// private key above: openssl rsa -pubout -outform DER -in k.pem | openssl md5 -c
+	expectedFingerprint := "73:61:a2:21:67:e0:df:be:7e:4b:93:1e:15:98:a5:b7"
+
+	expectedKeyID := strings.Join([]string{profile.TenancyID, profile.UserID, expectedFingerprint}, "/")
+
+	if expectedKeyID != profile.KeyID {
+		t.Fatalf("KeyID differs:\n\texpected: %s\n\treceived: %s", expectedKeyID, profile.KeyID)
+	}
+}
+
+func TestHTTPClient(t *testing.T) {
+	profile := GetDefaultAuthProfile(t)
+	profile.KeyID = "ocid1.tenancy.oc1..aaaaaaaaba3pv6wkcr4jqae5f15p2b2m2yt2j6rx32uzr4h25vqstifsfdsq/ocid1.user.oc1..aaaaaaaat5nvwcna5j6aqzjcaty5eqbb6qt2jvpkanghtgdaqedqw3rynjq/20:3b:97:13:55:1c:5b:0d:d3:37:d8:50:4e:c5:3a:34"
+
+	client := NewHTTPClient(context.TODO(), &profile)
```

Comment:
If it's possible without a lot of work, it might make sense to use a base RoundTripper which doesn't actually make a network request.